source/nsdlaccesshelper.cpp

```diff
@@ -31,7 +28,7 @@ uint8_t __nsdl_c_callback(struct nsdl_s *nsdl_handle,
                           sn_nsdl_capab_e nsdl_capab)
 {
     uint8_t status = 0;
-    M2MNsdlInterface  *interface = get_interface(nsdl_handle);
+    M2MNsdlInterface *interface = (M2MNsdlInterface*)sn_nsdl_get_context(nsdl_handle);
```

Comment:
I suppose a "assert(interface)" would make it sure the context is set properly on all cases.